Cython/Compiler/TypeSlots.py

```diff
@@ -381,6 +381,9 @@ def slot_code(self, scope):
         for method_name in self.alternatives:
             entry = scope.lookup_here(method_name)
             if entry and entry.is_special and entry.func_cname:
+                warn_once(entry.pos,
+                          f"{method_name} was removed in Python 3; use {self.method_name} instead",
+                          2)
```

Comment:
If the proposed alternative is defined, we could adapt the warning message to "…; ignoring it and using … instead".

Reply:
This would be fairly easy to do.